File: app/src/test/java/com/tripsta/tripstapoker/card/CardPresenterTest.java
package com.tripsta.tripstapoker.card;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

/**
 * Created by georgiossarris on 5/5/17.
 */
public class CardPresenterTest {

	@Mock
	private CardContract.View cardView;

	private CardPresenter cardPresenter;

	@Before
	public void setup(){
		MockitoAnnotations.initMocks(this);
		cardPresenter = new CardPresenter(cardView);
	}

	@Test
	public void initialize() throws Exception {
		cardPresenter.start();
		Mockito.verify(cardView).initializeGrid(CardPresenter.cards);
	}

	@Test
	public void showCard() throws Exception {
		int viewId = 123;
		cardPresenter.showCard(viewId);
		Mockito.verify(cardView).animateCard(viewId);
	}

}

File: app/src/test/java/com/sarris/poker/card/CardPresenterTest.kt
package com.sarris.poker.card

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.MockitoAnnotations

/**
 * Created by georgiossarris on 5/5/17.
 */
class CardPresenterTest {
    @Mock
    private val cardView: CardContract.View? = null

    private var cardPresenter: CardPresenter? = null

    @Before
    fun setup() {
        MockitoAnnotations.initMocks(this)
        cardPresenter = CardPresenter(cardView!!)
    }

    @Test
    @Throws(Exception::class)
    fun initialize() {
        cardPresenter!!.start()
        Mockito.verify(cardView)?.initializeGrid(CardPresenter.cards)
    }

    @Test
    @Throws(Exception::class)
    fun showCard() {
        val viewId = 123
        cardPresenter!!.showCard(viewId)
        Mockito.verify(cardView)?.animateCard(viewId)
    }
}